lib/Viroverse/ISLAWorksheet.pm

```diff
@@ -6,42 +6,55 @@ package Viroverse::ISLAWorksheet;
 use Moo;
 use Types::Standard qw< :types >;
 use Viroverse::Logger qw< :log >;
+use Viroverse::Types qw< ViroDBRecord >;
+use Excel::Writer::XLSX;
 
 =head1 NAME
 
-Viroverse::ISLAWorksheet - convert between a Viroverse::Result::Sample and an
-Excel spreadsheet for ISLA data
+Viroverse::ISLAWorksheet - convert between a L<ViroDB::Result::Sample> and an
+Excel spreadsheet for ISLA data.
 
 =head1 ATTRIBUTES
 
 =head2 model
 
-The Viroverse::Model::sample instance that this spreadsheet represents
+The L<ViroDB::Result::Sample> instance that this spreadsheet represents.
 
-=head1 METHODS
+=head2 sample_url
 
-=head2 make_xlsx
+A URL that maps to the Viroverse sample page.
 
-Generates an Excel spreadsheet (in xlsx format) that can be filled out with
-added data for the sample. Returns a FileHandle that has been seeked to the
-beginning.
+=head2 xlsx
+
+An Excel spreadsheet (in xlsx format) that can be filled out with added data
+for the sample. Returned as an IO::File; the seek position is not guaranteed.
 
 =cut
 
 has model => (
     is       => 'ro',
-    isa      => Object,
+    isa      => ViroDBRecord['Sample'],
     required => 1,
 );
 
-sub make_xlsx {
-    my ($self, $url) = @_;
+has sample_url => (
+    is       => 'ro',
+    isa      => Str,
+    required => 0,
+);
+
+has xlsx => (
+    is       => 'lazy',
+    isa      => Object,
```

Comment:
_checks docs_ You can likely use the `FileHandle` type exported from `Types::Standard` here.

Reply:
`Is::String` isn't a strict `FileHandle`

Reply:
This is moot if you change this property to just return the string of data but:

```perl
use Types::Standard qw< :types >;
use IO::String;

FileHandle->check(IO::String->new) ? print "OK\n" : print "NG\n";
```

Reply:
That might be the case but the type system was absolutely throwing an error that the type didn't match FileHandle.